48-kori-bustard/technical_specification.md

```diff
@@ -0,0 +1,171 @@
+
+> **IVA**, or **Independent Verification Address**, refers to some physical or digital
+> address, such as a mailing address or phone number, and is used to issue a code for
+> authentication.
+
+| Recipient    | Purpose                                  | Source Exists | Data Req'd |
+|--------------|------------------------------------------|---------------|------------|
+| Data Steward | LS Login doesn’t match what's registered | No            | User ID    |
+| User         | IVA invalidated                          | No            | User ID    |
+| User         | IVA verification code requested          | No            | User ID    |
+| Data Steward | IVA verification code requested          | No            | User ID    |
+| User         | IVA verification code transmitted        | No            | User ID    |
+| Data Steward | IVA verification code submitted by user  | No            | User ID    |
+
+
+_**Data Submission**_
+
+> For the "Research data upload completion" notification, the RDC email must be
+> retrievable from the File ID.
+
+| Recipient     | Purpose                           | Source Exists | Data Req'd |
+|---------------|-----------------------------------|---------------|------------|
+| Central DS    | *Metadata is ready for review     | No            | File ID    |
+| RD Controller | Research data upload completion   | Yes           | File ID    |
+| RD Submitter  | *Approval/rejection of submission | No            | User ID    |
+
+
+
+_**Data Request and Download**_
+
+> If there is a stored entity linking the request to both the dataset and user IDs, then
+> the request would be the only piece of information needed from the event.
+
+| Recipient    | Purpose                          | Source Exists | Data Req'd |
+|--------------|----------------------------------|---------------|------------|
+| DRR          | Request Created (Confirmation)   | No            | Dataset ID, User ID|
+| User (DACR)  | Request Created                  | No            | Dataset ID, User ID|
+| DRR          | Request Allowed                  | No            | Dataset ID, User ID|
+| User (DACR)  | Request Allowed                  | No            | Dataset ID, User ID|
+| DRR          | Request Denied                   | No            | Dataset ID, User ID|
+| User (DACR)  | Request Denied                   | No            | Dataset ID, User ID|
+| DRR          | *Dataset ready for download      | Yes           | Dataset ID, User ID|
+| DRR          | *Data access expiration reminder | No            | Dataset ID, User ID|
+| DRR          | *Data access expired             | No            | Dataset ID, User ID|
+
+_**Data Deletion**_
+| Recipient    | Purpose                     | Source Exists | Data Req'd |
+|--------------|-----------------------------|---------------|------------|
+| Data Steward | Deletion request received   | Yes           | File ID    |
+| Data Steward | *Deletion request fulfilled | Yes           | File ID    |
```

Comment:
This one would need additional functionality in the sense that NOS needs to be aware of all services that need to delete data and listen to all their deletion events + retention time for backups needs to be included in that calculation either way.

For now, I think using the request received one with a statement including our guaranteed date of deletion should suffice.

Reply:
Good insight, we'll keep it simple for now then.